python/experiments.py

```diff
@@ -38,9 +39,16 @@ def task_compile_benchmarks():
                 input_filepath=filepath,
                 output_dirpath=vivado_output_dirpath,
                 module_name=benchmark_name,
+                flags=''.join([f"-{elem} "for elem in benchmark_flags]) if benchmark_flags else '',
```

Comment:
I think this bit of processing is probably not useful -- can we just assume `benchmark['flags']` is a string, and pass the value of `benchmark['flags']` directly? and if it's not set, then pass `''`?

Reply:
So we could, but my thought behind this was that defining `flags` as a YAML list would allow for an arbitrarily large amount of flags to be added. 

I think it would become difficult to read the flags if they are defined as a multi-line string, but it's definitely possible.

Reply:
I understand your reasoning. We should still change it to a string. I will try to explain the engineering intuition here. Essentially, I always try to err on the side of providing a _coarser_ or _simpler_ API. This also usually means an uglier or more verbose API -- in this case uglier b/c the flags might be a long run-on string (though it doesn't necessarily have to be multi-line as you're saying -- it can just be a single line). A simpler API is better for a number of reasons:
1. With a simpler API, there's less guessing about what is happening under the hood. If we accepted a list of strings as our flags, then the user has to wonder: "are they appending nothing to each? are they appending `-` to each? are they separating them with spaces?" this forces them to actually look at the Python code to figure out what we're doing with the flags. If our flags are just a raw string, it's more obvious that they're just going to be appended directly to the command.
2. With a more complex API, you may make decisions that actually make certain things impossible. For example -- you forcibly append `-` to each flag. What if the user's flag doesn't take a `-` beforehand?
3. You can always make a simpler API more complex -- it's harder to simplify a complex API.

Reply:
Ahh ok I see, that makes a lot of sense. 

Thanks for the detailed explanation, I will change this section of the code to provide a simpler API.